### PR TITLE
feat(electron-test): move third hosted broker to self hosted broker

### PR DESCRIPTION
**Pull request by MaximoLiberata**
electron tests may present connection issues when try to connect to third hosted broker. The tests will fail for external issues not related to MQTT.JS. A self hosted broker will mitigate these issues.

### CODE REVIEW

**Comment by MaximoLiberata** [comment]
@axi92 Could you review this PR, too?

**Review by robertsLando — CHANGES_REQUESTED** [review]
I higly suggest to use aedes-cli instead:

https://github.com/mqttjs/MQTT.js/blob/2f8e73975de1c043b1691a564e120625cf84eb3d/web-test-runner.config.mjs#L8-L16

**Comment by MaximoLiberata** [comment]
@robertsLando I have a question, the `ServerLauncher` class/service is launched before each test starts. So far, the electron tests didn't fail, but I'm not sure if they will fail because duplicate server ports like node test issues.

Is there a way to close `aedes-cli` broker? I saw its the documentation, but I didn't find a method to close it

**Comment by robertsLando** [comment]
@MaximoLiberata `start` returns servers, broker instances so yeah you can close them:

https://github.com/moscajs/aedes-cli/blob/master/lib/cli.js#L420

**Comment by MaximoLiberata** [comment]
Thanks @robertsLando

File: electron-test/wdio.conf.ts [hunk 1]
```diff
@@ -116,7 +119,12 @@ export const config: Options.Testrunner = {
     // Services take over a specific job you don't want to take care of. They enhance
     // your test setup with almost no effort. Unlike plugins, they don't add new
     // commands. Instead, they hook themselves up into the test process.
-    services: ['electron'],
+    services: [
+        'electron',
+        [ServerLauncher, {
+            somsomeOption: true
```
Comment (robertsLando) [inline electron-test/wdio.conf.ts:125]: I think this is a typo? 
```suggestion
            someOption: true
```

File: electron-test/tsconfig.json [hunk 2]
```diff
@@ -19,7 +19,6 @@
         "allowImportingTsExtensions": true,
         "resolveJsonModule": true,
         "isolatedModules": true,
-        "strict": true,
```
Comment (robertsLando) [inline electron-test/tsconfig.json:22]: @axi92 This affects only the electron test project not the MQTTjs project so not a big deal IMO

**Review by robertsLando — CHANGES_REQUESTED** [review]

**Comment by robertsLando** [comment]
@MaximoLiberata Sorry for the late review but I'm on holiday, I will be back on the end of August :)

File: electron-test/tsconfig.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (MaximoLiberata) [reply electron-test/tsconfig.json]: Done. Please check it.

Comment (MaximoLiberata) [reply electron-test/tsconfig.json]: Done. Please check it.

File: electron-test/wdio.conf.ts [hunk 3]
```diff
@@ -116,7 +119,12 @@ export const config: Options.Testrunner = {
     // Services take over a specific job you don't want to take care of. They enhance
     // your test setup with almost no effort. Unlike plugins, they don't add new
     // commands. Instead, they hook themselves up into the test process.
-    services: ['electron'],
+    services: [
+        'electron',
+        [ServerLauncher, {
+            someOption: true
```
Comment (axi92) [inline electron-test/wdio.conf.ts:125]: I don't think we need the options, do we?
I guess you got it from that [example](https://webdriver.io/docs/customservices/#import-service-from-module)?

Comment (MaximoLiberata) [reply electron-test/wdio.conf.ts:125]: Yes, I was following the instructions of the example. We can remove the option to reduce code or let it.

@axi92 Which option do you think is best? I'm according with both.

Comment (robertsLando) [reply electron-test/wdio.conf.ts:125]: I agree this can be removed as it seems to be just a placeholder option put there as an example

**Review by robertsLando — CHANGES_REQUESTED** [review]
Just came back from holidays, sorry for the late review! Everything looks good to me, just remove the useless option and I will merge this, thanks 🙏🏼

File: electron-test/wdio.conf.ts (hunk 3 above)
Comment (MaximoLiberata) [reply electron-test/wdio.conf.ts]: Done.

**Comment by MaximoLiberata** [comment]
thanks @robertsLando and @axi92 for review 🙏🏼

**Review by robertsLando — APPROVED** [review]

**Pull request merged**